src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -691,36 +699,54 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
 // @todo add ascii diagramm for memory mapping / reuse
 void MemoryInput::createPrimitive() {
-    MemoryInputBase::createPrimitive();
     if (haveSubgraph()) {
-        OPENVINO_ASSERT(getOriginalInputsNumber() == subGraph->inputsNumber(),
-                        "Number of node inputs must be equal the number of inner graph's inputs: ",
-                        getOriginalInputsNumber(),
-                        " != ",
-                        subGraph->inputsNumber());
-
-        std::vector<MemoryPtr> inputMemory;
-        for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
-            auto srcEdgeMem = getSrcMemoryAtPort(i);
-            // create a separate input memory objects instead of share them. avoid data corruption.
-            auto mem = std::make_shared<Memory>(getEngine(), srcEdgeMem->getDescPtr(), srcEdgeMem->getMemoryBlock());
-            subgraphMemoryPtrs.push_back(mem);
-            inputMemory.emplace_back(std::move(mem));
+        CPU_NODE_ASSERT(getParentEdges().size() == subGraph->inputsNumber(),
+                        "Number of node inputs must be equal the number of inner graph's inputs");
+
+        for (size_t i = 0; i < subGraph->inputsNumber(); i++) {
```

Comment:
Please note that when the `subgraphMemoryPtrs` are being used, the traversal goes using `getOriginalInputsNumber()`. Probably it does make sense to align these two loops.

Reply:
Updated